specutils/helcorr.py

```diff
@@ -0,0 +1,343 @@
+# coding: utf-8
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+""" Heliocentric velocity corrections """
+
+from __future__ import division, print_function
+
+# This code has been modified from legacy codes. Credits are chronologically
+# attributable to: Sergey Koposov, Kochukho, Kudryavtsev, W. Landsman, 
+# Chris McCarthy, and Jeff Valenti.
+
+# Third-party
+import numpy as np
+
+import astropy.constants as constants
+import astropy.coordinates as coord
+import astropy.time
+import astropy.units as u
+
+__all__ = ["helcorr", "baryvel"]
+
+def baryvel(dje):
+    """
+    Calculates the heliocentric and barycentric velocity components of Earth.
+
+    Parameters
+    ----------
+    dje : `~astropy.time.Time` or float
+        The Julian ephemeris date.
+
+    Returns
+    -------
+    dvelh : `~astropy.units.Quantity`
+        The heliocentric velocity components in (X, Y, Z) coordinates.
+
+    dvelb : `~astropy.units.Quantity`
+        The barycentric velocity components in (X, Y, Z) coordinates.
+    """
+
+    if isinstance(dje, astropy.time.Time):
```

Comment:
For ducktyping I would use this piece of code (not doing type checks, such as `type` or `isinstance`:

```python
if hasattr(dje, 'jd'):
    jd = dje.jd
else:
   jd = dje
```